Support custom markers (#76)
elastic/crd-ref-docs: README.md

      package: sigs.k8s.io/gateway-api/apis/v1beta1
      link: https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.SecretObjectReference
```

### Advanced Features

#### Custom Markers

You can add custom markers to your CRD types to provide additional information in the generated documentation. 
For example, you can add a `hidefromdoc` marker to indicate that a type is hide from the documentation.

```yaml
processor:
 ignoreGroupVersions:
   - "GVK"
 ignoreTypes:
   - "Embedded[2-4]$"
 ignoreFields:
   - "status$"
   - "TypeMeta$"
 customMarkers:
   - name: "hidefromdoc"
     target: field

render:
 kubernetesVersion: 1.25
 knownTypes:
   - name: SecretObjectReference
     package: sigs.k8s.io/gateway-api/apis/v1beta1
     link: https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.SecretObjectReference
```

> You can then add the `hidefromdoc` marker to the field you want to hide from the documentation.

Reviewer: ```suggestion
You can then add the `hidefromdoc` marker to the field you want to hidden from the documentation.
```

```go
type Embedded1 struct {
	Embedded2 `json:",inline"`
	// +hidefromdoc
	E         string `json:"e,omitempty"`
	EmbeddedX `json:",inline"`
}
```

> Then update the templates to render the custom markers. You can find the demo from [here](./test/templates/markdown/type.tpl).

Reviewer: ```suggestion
Then update the templates to render the custom markers. You can find an example [here](./test/templates/markdown/type.tpl).
```
